src/libelfmaster.c

```diff
@@ -2518,7 +2518,7 @@ elf_section_iterator_next(struct elf_section_iterator *iter,
  * Secure ELF loader.
  */
 bool
-elf_open_object(const char *path, struct elfobj *obj, uint64_t load_flags,
```

Comment:
We only declare the type as opaque when we aren't accessing the struct directly, which is how programmers will use the interface, they should never have to access elfobj_t directly. However the function `elf_open_object` accesses it directly (i.e. obj->member) which is ok since it's a libelfmaster function. But we don't want to declare it as `elfobj_t *obj` we want to keep it as `struct elfobj *obj`. This is how we handle opaque types in the programming style of libelfmaster.

Reply:
ahh I see